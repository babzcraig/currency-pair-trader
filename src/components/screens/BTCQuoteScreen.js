import React from 'react';
import PropTypes from 'prop-types';
import { getBTCQuote } from '../../utils';


const BTCQuoteScreen = ({lastPrice, amountToBuy}) => {
  const btcQuote = getBTCQuote(lastPrice, amountToBuy);

  return (
    <React.Fragment>
      <h2>For</h2>
      <div>
        <input type="number" placeholder="BTC" disabled/>
      </div>
      <div>
        <input type="number" placeholder="Quoted Price" value={btcQuote} disabled/>
      </div>
    </React.Fragment>
  )
}

BTCQuoteScreen.propTypes = {
  lastPrice: PropTypes.number,
  amountToBuy: PropTypes.string
}

export default BTCQuoteScreen;
