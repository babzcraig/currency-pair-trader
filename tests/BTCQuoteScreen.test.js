import React from 'react';
import ReactDOM from 'react-dom'
import BTCQuoteScreen from '../src/components/screens/BTCQuoteScreen'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("<BTCQuoteScreen />", () => {
  const wrapper = shallow(<BTCQuoteScreen />);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BTCQuoteScreen />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('contains two input fields', () => {
    expect(wrapper.find('input').length).toBe(2)
  });

  it('contains an h2 with a Text of "For', () => {
    expect(wrapper.find('h2').text()).toBe('For')
  });

  it('matches the snapshot', () => {
    const tree = shallow(<BTCQuoteScreen/>);
    expect(toJson(tree)).toMatchSnapshot()
  })
})


