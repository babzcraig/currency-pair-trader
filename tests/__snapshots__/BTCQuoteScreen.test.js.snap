// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BTCQuoteScreen /> matches the snapshot 1`] = `
<React.Fragment>
  <h2
    className="trade-section"
  >
    For
  </h2>
  <div>
    <input
      disabled={true}
      placeholder="BTC"
      type="number"
    />
  </div>
  <div>
    <input
      disabled={true}
      placeholder="Quoted Price"
      type="number"
    />
  </div>
</React.Fragment>
`;
